compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -320,7 +341,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let tcx = self.tcx;
         let expected_inner = match unop {
             hir::UnOp::UnNot | hir::UnOp::UnNeg => expected,
-            hir::UnOp::UnDeref => NoExpectation,
+            hir::UnOp::UnDeref => NoExpectation(expected.get_coercion_ctxt()),
```

Comment:
I don't think we want to consider this a coercion site. This corresponds to an expression like `*foo`, and it doesn't make sense to "coerce" `foo` to .. anything.